include/cudaq/Support/Version.h

```diff
@@ -0,0 +1,20 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+namespace cudaq {
+
+// TODO: const char *getVersion();
+
+/// A generic bug report message.
+constexpr const char *bugReportMsg =
```

Comment:
Should this be a `static`? Many other strings variables in include/cudaq/Optimizer/CodeGen/QIRFunctionNames.h are static.

Reply:
In this particular case it shouldn't be necessary because `constexpr` implies `const`, which on global/namespace scope implies `static` (i.e., internal linkage). This means that every translation unit including this header gets its own copy of the variable, and that adding `static` wouldn't make a difference.

However, it might be good to add `inline`, and also substituting `static` by `inline` in that file. This is something that was added in C++17 and is considered good practice by some people. The `inline` keyword means "This variable might be defined in multiple translation units (the ones which include the header); all those definitions are identical; merge them into one definition at link time." This would be required if we wanted the address of this constant to be the same everywhere, or if we wanted to save some space (really negligible) by having only one copy of the string. (Again, not really necessary in this case, because the `Version.h` header is only added once per tool.)

Reply:
OK, that sounds reasonable. I double checked it by #including Version.h into a few different test files and did not see any link issues, as long as `constexpr` was included.  And I agree the hypothetical duplicate definitions would be negligible anyway. You can disregard this comment then.